hajarel

hajar98 opened the pull request.

hajar98 committed:
hajarel

Robright20 reviewed (commented):
.

Robright20 reviewed (commented):
#2 Hello Hajar!

We are very glad that you were able to pass the test. I take the time to check your code 
and I'll write down my review.

First, I have noticed some good things that I want to mention:
- the use of the ternary,
- the "strict equal operator"
- you had a good understanding of the instructions
- the program works for all the basics tests case
- you had a good deduction skill
- good understanding of the basics(primitives, functions, ....).
- the usage of Object methods like (slice, join, split)
- you did well by testing your code with console.log

Second, here are some suggestions to improve your code.
- while writing your code try to follow a good coding convention and
keep your code more readable(here is a [link](https://www.w3schools.com/js/js_conventions.asp) and easy to debug.
- don't forget your debug statements(e.g console.log) after you complete your task.
- you didn't understand why we put `throw Error('No tests !')` and `use strict`
- you can use arrow functions for functions with one statement and write them in
in a more concise way.

hajar98 commented:
Good evening sir,I hope you are doing well. Thank you for giving me this
chance to prove myself. I will take your advice in consideration to improve
my code. have a good night.



Le jeu. 24 déc. 2020 à 17:47, Robert Bright <notifications@github.com> a
écrit :

> *@Robright20* commented on this pull request.
>
> #2 <https://github.com/Robright20/js-training/pull/2> Hello Hajar!
>
> We are very glad that you were able to pass the test. I take the time to
> check your code
> and I'll write down my review.
>
> First, I have noticed some good things that I want to mention:
>
>    - the use of the ternary,
>    - the "strict equal operator"
>    - you had a good understanding of the instructions
>    - the program works for all the basics tests case
>    - you had a good deduction skill
>    - good understanding of the basics(primitives, functions, ....).
>    - the usage of Object methods like (slice, join, split)
>    - you did well by testing your code with console.log
>
> Second, here are some suggestions to improve your code.
>
>    - while writing your code try to follow a good coding convention and
>    keep your code more readable(here is a link
>    <https://www.w3schools.com/js/js_conventions.asp> and easy to debug.
>    - don't forget your debug statements(e.g console.log) after you
>    complete your task.
>    - you didn't understand why we put throw Error('No tests !') and use
>    strict
>    - you can use arrow functions for functions with one statement and
>    write them in
>    in a more concise way.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Robright20/js-training/pull/2#pullrequestreview-558684127>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APAZ36AE4PIUAFY2QD2KHA3SWN5D5ANCNFSM4VHQJQBA>
> .
>

-- 
-------------------------------------------------- 
*Disclaimer (inserted 
automatically):*



* Ce message ainsi que les éventuelles pièces jointes 
constituent une correspondance privée et confidentielle à l'attention 
exclusive du (des) destinataire (s) designé (s) ci-dessus. Si vous n'êtes 
pas le destinataire du présent message ou une personne susceptible de 
pouvoir le lui délivrer, il vous est signalé que toute divulgation, 
distribution ou copie de cette transmission est strictement interdite. Si 
vous avez reçu ce message par erreur, nous vous remercions d'en informer 
l'expéditeur par téléphone ou de lui retourner le présent message, puis 
d'effacer immédiatement ce message de votre système.
L'intégrité des 
messages n'étant pas assurée sur Internet, l'ENSEM ne saurait être tenu 
responsable si ce message s'avérait modifié ou falsifié. Autrement, si 
d'autres messages vous sont parvenus sans cet avertissement, il serait fort 
probable qu'ils soient manipulés par d'autres systèmes ou personnes 
externes.


* This e-mail and any attachments is a confidential 
correspondence intended only for use of the individual or entity named 
above. If you are not the intended recipient or the agent responsible for 
delivering the message to the intended recipient, you are hereby notified 
that any disclosure, distribution or copying of this communication is 
strictly prohibited. If you have received this communication in error, 
please notify the sender by phone or by replying this message, and then 
delete this message from your system.Thank you for your help. 
Messages are 
susceptible to alteration. ENSEM shall not be liable for the message if 
altered, changed or falsified. Otherwise, if other messages are sent to you 
without this "disclaimer", they are very likely to be manipulated by other 
systems or external persons. So, be careful.
-------------------------------------------------- 

lsoulima reviewed (commented):
Salut Hajar, je suis heureuse de voir plus de femmes au monde du coding .Alors en révisant tes codes, je vois que tu as assez de connaissances en Javascript sauf que je veux mettre en highlight les points mentionnés par mon collègue surtout respecter les normes du coding et de faire attention aux details ...
BON COURAGE

Voilà le résultat de tests que j'ai fait. (Ce qui est coche c'est qui a marché)
- [x] abs.js
- [x] add.js
- [x] arr.js
- [x] capitalize.js
- [ ] circular.js
- [x] concat-str.js
- [x] cut.js
- [ ] escape-str.js
- [x] get-first.js
- [x] get-last.js
- [x] get-length.js
- [ ] get.js
- [x] id.js
- [ ] is-positive.js
- [ ] jaden-case.js
- [ ] keep.js
- [x] max.js
- [x] min.js
- [ ] multiply-recursive.js
- [ ] multiply.js
- [ ] obj.js
- [ ] primitives.js
- [ ] repeat.js
- [ ] same-sign.js
- [x] sentence.js
- [x] sign.js
- [x] sub.js
- [x] total.js
- [x] whisper.js
- [x] words.js
- [x] yell.js

hajar98 commented:
Bonsoir , merci déjà pour l'opportunité que vous m'aviez donné, merci aussi
pour donner de votre temps pour faire la correction , je vais prendre en
considération les points que vous avez mentionné. Je veux mentionner que
 j'ai pas pu continuer la totalité du test  ,car j'ai galéré avec la
connexion.Merci encore une fois et bonne soirée.

Le jeu. 24 déc. 2020 à 19:27, lsoulima <notifications@github.com> a écrit :

> *@lsoulima* commented on this pull request.
>
> Salut Hajar, je suis heureuse de voir plus de femmes au monde du coding
> .Alors en révisant tes codes, je vois que tu as assez de connaissances en
> Javascript sauf que je veux mettre en highlight les points mentionnés par
> mon collègue surtout respecter les normes du coding et de faire attention
> aux details ...
> BON COURAGE
>
> Voilà le résultat de tests que j'ai fait. (Ce qui est coche c'est qui a
> marché)
>
>    - abs.js
>    - add.js
>    - arr.js
>    - capitalize.js
>    - circular.js
>    - concat-str.js
>    - cut.js
>    - escape-str.js
>    - get-first.js
>    - get-last.js
>    - get-length.js
>    - get.js
>    - id.js
>    - is-positive.js
>    - jaden-case.js
>    - keep.js
>    - max.js
>    - min.js
>    - multiply-recursive.js
>    - multiply.js
>    - obj.js
>    - primitives.js
>    - repeat.js
>    - same-sign.js
>    - sentence.js
>    - sign.js
>    - sub.js
>    - total.js
>    - whisper.js
>    - words.js
>    - yell.js
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Robright20/js-training/pull/2#pullrequestreview-558699562>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APAZ36CSPCPAUUU4GHTXKPDSWOIYVANCNFSM4VHQJQBA>
> .
>

-- 
-------------------------------------------------- 
*Disclaimer (inserted 
automatically):*



* Ce message ainsi que les éventuelles pièces jointes 
constituent une correspondance privée et confidentielle à l'attention 
exclusive du (des) destinataire (s) designé (s) ci-dessus. Si vous n'êtes 
pas le destinataire du présent message ou une personne susceptible de 
pouvoir le lui délivrer, il vous est signalé que toute divulgation, 
distribution ou copie de cette transmission est strictement interdite. Si 
vous avez reçu ce message par erreur, nous vous remercions d'en informer 
l'expéditeur par téléphone ou de lui retourner le présent message, puis 
d'effacer immédiatement ce message de votre système.
L'intégrité des 
messages n'étant pas assurée sur Internet, l'ENSEM ne saurait être tenu 
responsable si ce message s'avérait modifié ou falsifié. Autrement, si 
d'autres messages vous sont parvenus sans cet avertissement, il serait fort 
probable qu'ils soient manipulés par d'autres systèmes ou personnes 
externes.


* This e-mail and any attachments is a confidential 
correspondence intended only for use of the individual or entity named 
above. If you are not the intended recipient or the agent responsible for 
delivering the message to the intended recipient, you are hereby notified 
that any disclosure, distribution or copying of this communication is 
strictly prohibited. If you have received this communication in error, 
please notify the sender by phone or by replying this message, and then 
delete this message from your system.Thank you for your help. 
Messages are 
susceptible to alteration. ENSEM shall not be liable for the message if 
altered, changed or falsified. Otherwise, if other messages are sent to you 
without this "disclaimer", they are very likely to be manipulated by other 
systems or external persons. So, be careful.
-------------------------------------------------- 
